process/smartContract/process.go

```diff
@@ -1430,6 +1415,11 @@ func (sc *scProcessor) createSmartContractResults(
 			createdAsyncCallBack = true
 		}
 
+		if result.CallType == vmcommon.AsynchronousCall {
+			lastArgAsGasLocked := "@" + hex.EncodeToString(big.NewInt(0).SetUint64(outputTransfer.GasLocked).Bytes())
+			result.Data = append(result.Data, []byte(lastArgAsGasLocked)...)
```

Comment:
Is this required for backwards compatibility?

`result` already has the field `GasLocked`. I recommend setting `result.GasLocked`. Then, if needed, changing `getAsyncCallGasLockFromTxData()` so that it casts the transaction into a SCR and getting the locked gas from the SCR, if `callType == AsynchronousCall`.

Are there other reasons which require sending the locked gas through `Data`?

Reply:
result does not have field GasLocked. this is the only way to resolve without creating a new smartContractResult data structure